tox.ini

```diff
@@ -15,75 +12,70 @@ src_path = {toxinidir}/src/
 tst_path = {toxinidir}/tests/
 
 [testenv]
-passenv =
-    PYTHONPATH
-    CHARM_BUILD_DIR
-    MODEL_SETTINGS
-    KUBECONFIG
-setenv =
-    PYTHONPATH = {toxinidir}:{toxinidir}/lib:{[vars]src_path}
-    PYTHONBREAKPOINT=ipdb.set_trace
-    PY_COLORS=1
+passenv = 
+	PYTHONPATH
+	CHARM_BUILD_DIR
+	MODEL_SETTINGS
+	KUBECONFIG
+setenv = 
+	PYTHONPATH = {toxinidir}:{toxinidir}/lib:{[vars]src_path}
+	PYTHONBREAKPOINT=ipdb.set_trace
+	PY_COLORS=1
 
 [testenv:update-requirements]
-allowlist_externals =
-    bash
-    find
-    pip-compile
-    xargs
+allowlist_externals = 
+	bash
+	find
+	pip-compile
+	xargs
 commands = 
-    ; we must preserve the order of compilation, since each *.in file depends on some *.txt file.
-    ; For example, requirements-unit.in depends on requirements.txt and we must compile first
-    ; requirements.txt to ensure that requirements-unit.txt get the same dependency as the requirements.txt
-    bash -c 'for pattern in "requirements.in" "requirements-fmt.in" "requirements*.in"; do find . -type f -name "$pattern" -exec bash -c "cd \$(dirname "{}") && pip-compile --resolver=backtracking \$(basename "{}")" \;; done'
-deps =
-    pip-tools
-    # Pin due to https://github.com/jazzband/pip-tools/issues/2131
-    pip==24.2
+	bash -c 'for pattern in "requirements.in" "requirements-fmt.in" "requirements*.in"; do find . -type f -name "$pattern" -exec bash -c "cd \$(dirname "{}") && pip-compile --resolver=backtracking \$(basename "{}")" \;; done'
+deps = 
+	pip-tools
+	pip==24.2
 description = Update requirements files by executing pip-compile on all requirements*.in files, including those in subdirs.
 
 [testenv:fmt]
-commands =
-    isort {[vars]all_path}
-    black {[vars]all_path}
-deps =
-    -r requirements-fmt.txt
+commands = 
+	isort {[vars]all_path}
+	black {[vars]all_path}
+deps = 
+	-r requirements-fmt.txt
 description = Apply coding style standards to code
 
 [testenv:lint]
-commands =
-    # uncomment the following line if this charm owns a lib
-    # codespell {[vars]lib_path}
-    codespell {toxinidir}/. --skip {toxinidir}/./.git --skip {toxinidir}/./.tox \
-      --skip {toxinidir}/./build --skip {toxinidir}/./lib --skip {toxinidir}/./venv \
-      --skip {toxinidir}/./.mypy_cache \
-      --skip {toxinidir}/./icon.svg --skip *.json.tmpl
-    # pflake8 wrapper supports config from pyproject.toml
-    pflake8 {[vars]all_path}
-    isort --check-only --diff {[vars]all_path}
-    black --check --diff {[vars]all_path}
-deps =
-    -r requirements-lint.txt
+commands = 
+	codespell {toxinidir}/. --skip {toxinidir}/./.git --skip {toxinidir}/./.tox \
+	--skip {toxinidir}/./build --skip {toxinidir}/./lib --skip {toxinidir}/./venv \
```

Comment:
(not a blocker): I 'm not a big fan of the formatting this introduces (everything at the same level). If that's not configurable, we can proceed as is.

Reply:
This is really the default that one would have when using parsing and writing using python libraries, so I'd argue it is better. It is nevertheless required if we want also to enable a way to programatically manage tox files to enable automation. I'd rather pick formatting that can are friendly to programatically processes